drivers/lora/sx126x_stm32wl.c

```diff
@@ -45,6 +45,53 @@ void sx126x_dio1_irq_disable(struct sx126x_data *dev_data)
 	irq_disable(DT_INST_IRQN(0));
 }
 
+void sx126x_set_tx_params(int8_t power, RadioRampTimes_t ramp_time)
+{
+	uint8_t buf[2];
+
+	if (DT_INST_ENUM_IDX(0, power_amplifier_output) == 0) {
```

Comment:
This is my only concern with the implementation, it is a brittle way of checking the enum value.
I have opened #56874 which will hopefully allow writing this in a clearer fashion.
`if (DT_INST_ENUM_HAS_VALUE(0, power_amplifier_output, rfo_lp))`

Reply:
I generally agree with your suggestion and associated PR, but the usage here IMHO is robust.

1. A DTS enum will always have at least one value which will match the first conditional outcome. For any other value, the alternative code path acts as a catch-all mimicking the behavior before this PR.
2.  The added property is marked `required`, so any DTS configurations activating this portion of the SX126x driver with `CONFIG_LORA_STM32WL_SUBGHZ_RADIO` through `DT_HAS_ST_STM32WL_SUBGHZ_RADIO_ENABLED` will fail the DTC process, should the property be missing or its value or not match one of the two defined values.
3. From a quick grep I see at least 9 other comparable uses of `DT_ENUM_IDX` and its variations throughout the Zephyr codebase. I think those and this PR's use of which should be addressed separately.

Reply:
> I generally agree with your suggestion and associated PR, but the usage here IMHO is robust.

I disagree that it can be considered robust when simply swapping the order of `rfo-lp` and `rfo-hp` in the binding file will invert the behaviour of the code.

Reply:
Naturally, enum ordering is important. This PR adds the enum property and the logic handling it in a single commit, why would the order in the binding definition be subject to change without consideration for the code handling it?

Also, how is this different from the dozens of other uses of `DT_ENUM_IDX` and co. where the enum index is directly operated on, such as  non-successive integer enums, or enums used indirectly with a matching C-source enum? (Which with the same argument may also deviate in order)

I'm all for decreasing ambiguity in DTS enum value operations, but PRs shouldn't suddenly be blocked if it already occurs several times in the existing codebase.

In my experience, creating dependencies between functionally unrelated PRs unnecessarily increases reviewer spread and slows down integration agility.

Reply:
> Naturally, enum ordering is important. This PR adds the enum property and the logic handling it in a single commit, why would the order in the binding definition be subject to change without consideration for the code handling it?

If enum values are important, they should be of the form: `enum { VAL_A = 1, VAL_B = 2 };`. The problem here is the complete lack of any link between how the enum is generated (binding file) and where it is used (the comparison). This is the exact reason why enum support was improved as a result of #21273.

> Also, how is this different from the dozens of other uses of DT_ENUM_IDX and co. where the enum index is directly operated on, such as non-successive integer enums, or enums used indirectly with a matching C-source enum? (Which with the same argument may also deviate in order)

Just because it hasn't been caught by other reviewers doesn't make the solution any more correct IMO.

> I'm all for decreasing ambiguity in DTS enum value operations, but PRs shouldn't suddenly be blocked if it already occurs several times in the existing codebase.

If it is time critical for this to be merged for some reason, there are other options that aren't quite as ergonomic but don't have the same problem.
```
enum {
   PA_RFO_LP,
   PA_RFO_HP
} dt_inst_power_mode = UTIL_CAT(PA_, DT_INST_STRING_UPPER_TOKEN(0, power_amplifier_output));
...
if (dt_inst_power_mode == PA_RFO_LP) {}
```

Reply:
> If it is time critical for this to be merged for some reason, there are other options that aren't quite as ergonomic but don't have the same problem.

I just want to avoid it getting mothballed by dependency-chaining it to to an issue which might evoke longer discussion, when it solves a real problem with a hardware feature for a supported board that *seems* to be operational, but really is set to a non-functional and dangerous (for the longevity of the hardware) configuration.

Reply:
I like the use of `DT_INST_STRING_UPPER_TOKEN()`, as shown above.

Reply:
OK, after some consideration I think it's a good compromise.